src/Microsoft.AspNet.Mvc.TagHelpers/OptionTagHelper.cs

```diff
@@ -86,7 +86,7 @@ public class OptionTagHelper : TagHelper
                     // Select this <option/> element if value attribute or content matches a selected value. Callers
                     // encode values as-needed while executing child content. But TagHelperOutput itself
                     // encodes attribute values later, when GenerateStartTag() is called.
-                    var text = await context.GetChildContentAsync();
+                    var text = (await context.GetChildContentAsync()).GetContent();
```

Comment:
my bad: I shouldn't have called `GetChildContentAsync()` at all when `Value != null`.  agree this should be fixed.

just don't try to fit it all into a ternary expression :smile_cat: